src/test/pages/EventDetails.test.tsx

```diff
@@ -0,0 +1,229 @@
+  });
+
+  it("shows a map and address details if location, latitude and longitude are specified, and event is not tagged as online", async () => {
+    const mapLocation = {
+      address: { addressLine1: "Mock street", town: "Mock town", postalCode: "ABC 123" },
+      latitude: 53.95038292218263,
+      longitude: -1.0510007751299424,
+    };
+    await setupTest({
+      role: "STUDENT",
+      event: { ...mockEvent, tags: ["booster", "student"], location: mapLocation },
+    });
+    const { map, location } = getFields();
+    expect(map()).toBeInTheDocument();
+    expect(location()).toHaveTextContent(formatAddress(mapLocation));
+  });
+
+  it("shows location as online if event is tagged as virtual", async () => {
+    await setupTest({ role: "STUDENT", event: { ...mockEvent, tags: ["virtual"] } });
+    const { location } = getFields();
+    expect(location()).toHaveTextContent("Online");
+  });
+
+  it("shows private event badge if event is private", async () => {
+    await setupTest({ role: "STUDENT", event: { ...mockEvent, privateEvent: true } });
+    const { privateBadge } = getFields();
+    expect(privateBadge()).toBeVisible();
+  });
+
+  it("does not show private event badge if even is not private", async () => {
+    await setupTest({ role: "STUDENT", event: { ...mockEvent, privateEvent: false } });
+    const { privateBadge } = getFields();
+    expect(privateBadge()).not.toBeInTheDocument();
+  });
+
+  it.each(["ADMIN", "EVENT_MANAGER", "CONTENT_EDITOR"] as TestUserRole[])(
+    "if user is %s, a google calendar button shows and can be clicked",
+    async (role) => {
+      const mockGoogleCalendarTemplate = jest.fn();
+      jest.spyOn(eventServices, "googleCalendarTemplate").mockImplementation(mockGoogleCalendarTemplate);
+      await setupTest({ role: role, event: mockEvent });
+      const { googleCalendarButton } = getFields();
+      expect(googleCalendarButton()).toBeVisible();
+      await userEvent.click(googleCalendarButton() as HTMLElement);
+      expect(mockGoogleCalendarTemplate).toHaveBeenCalled();
+    },
+  );
+
+  it("if user is STUDENT, a google calendar button does not show", async () => {
+    await setupTest({ role: "STUDENT", event: mockEvent });
+    const { googleCalendarButton } = getFields();
+    expect(googleCalendarButton()).not.toBeInTheDocument();
+  });
+
+  it("shows event date and time, and message if event was in the past", async () => {
+    const startDate = new Date(new Date().setMonth(new Date().getMonth() - 2));
+    const endDate = new Date(startDate.getTime() + 2 * 60 * 60 * 1000);
+    const pastEvent = { ...mockEvent, date: startDate, endDate: endDate };
+    await setupTest({
+      role: "STUDENT",
+      event: pastEvent,
+    });
+    const { eventDate } = getFields();
+    const eventDateText = `${FRIENDLY_DATE_AND_TIME.format(startDate)} — ${TIME_ONLY.format(endDate)}`;
+    expect(eventDate()).toHaveTextContent(eventDateText);
```

Comment:
One problem with now having these as functions is that it's more inefficient as you're calling the findBy, queryBy, etc methods multiple times in the same test when one would suffice. The fix is easy: saving the result in a constant in the relevant tests. This happens in a few other tests in this file, not only here.